Refactor

mlakatkou opened the pull request:
To ensure compatibility with new versions of dependencies and improve development experience.

mlakatkou committed:
[refaktor] Update dependencies

To ensure compatibility with new versions of dependencies and improve development experience.

mlakatkou committed:
Merge remote-tracking branch 'origin/main' into refactor

AlexKlimenkov commented on package.json:
```diff
@@ -1,33 +1,39 @@
   "scripts": {
     "start": "react-scripts start",
     "build": "react-scripts build",
-    "test": "react-scripts test --env=jsdom",
```
please keep `--env=jsdom`, unless it's no longer needed.

AlexKlimenkov commented on public/index.html:
```diff
@@ -1,22 +1,43 @@
+    <meta name="theme-color" content="#000000" />
+    <meta
+      name="description"
+      content="Web site created using create-react-app"
```
add an appropriate description

AlexKlimenkov commented on public/index.html:
```diff
@@ -1,22 +1,43 @@
+      work correctly both with client-side routing and a non-root public URL.
+      Learn how to configure a non-root public URL by running `npm run build`.
+    -->
+    <title>React App</title>
```
please the title, `DHTMLX Event Calendar with React` or similar

AlexKlimenkov commented on src/EventCalendar.jsx:
```diff
@@ -0,0 +1,18 @@
+			date: props.date,
+		});
+
+		return () => (container.current.innerHTML = "");
```
need to do `eventCalendar.destructor()`  on unmount https://docs.dhtmlx.com/eventcalendar/api/methods/js_eventcalendar_destructor_method/

mlakatkou committed:
[update] Add --env=jsdom flag to package.json file

To utilize the JSDOM environment for React testing.

mlakatkou committed:
[update] Change title in the index.html

To improve the page's relevance.

mlakatkou committed:
[update] Change description in the index.html

To improve the page's relevance.

mlakatkou committed:
[update] Add destructor method

To remove all HTML elements of the Event Calendar and detach all related events after the Event Calendar component has been unmounted from the DOM.
